base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/error.rs

```diff
@@ -66,4 +66,6 @@ pub enum HorizonSyncError {
     ConversionError(String),
     #[error("MerkleMountainRangeError: {0}")]
     MerkleMountainRangeError(#[from] MerkleMountainRangeError),
+    #[error("The header height changed during validation")]
+    HeaderHeightChanged,
```

Comment:
Suggestion: A concurrency error might be more general. The problem with the naming here is that it sounds like the height field of a single header has changed. Perhaps also `ChainHeightChanged` or `TipChanged`
```suggestion
     ConcurrencyError,
```